apis/config/v1beta1/configuration_types.go

```diff
@@ -199,6 +202,13 @@ type WaitForPodsReady struct {
 	RequeuingTimestamp *RequeuingTimestamp `json:"requeuingTimestamp,omitempty"`
 }
 
+type MultiKueue struct {
+	// GCTimeout defines the timeout between two consecutive garbage collection runs.
+	// Defaults to 1min. If 0, the garbage collection is disabled.
+	// +optional
+	GCTimeout *metav1.Duration `json:"gcTimeout,omitempty"`
```

Comment:
Typically I prefer not to expose configuration options we are unsure of being needed. One reason is making things configurable we risk that users without good understanding would tweak this option (by disabling, setting too often or too rare) running into issues, and require OnCall help. Another is that maybe in the future we prefer to have an adaptative interval rather than a constant one. This is still alpha, so I think it is ok to wait for feedback from Alpha testers and see if they need it configurable.

This is analogous to the core k8s [gcCheckPeriod](https://github.com/kubernetes/kubernetes/blob/eb0fcf9e216f83e200e1c04e24d0484394da6b86/pkg/controller/podgc/gc_controller.go#L50C2-L50C15), which is not exposed to users. 

Still. I see some benefits, so I'm fine with this configuration option if @alculquicondor or @tenzen-y give thumbs up for the option.

Reply:
In case this is exposed just for testing we can have two strategies, as in the code k8s using `NewPodGCInternal` to override the defaults, or as we do for example with there parameters: https://github.com/kubernetes-sigs/kueue/blob/b7fc93f5ba42ebbcd54345702be24c3f76d7ea26/pkg/controller/admissionchecks/provisioning/controller.go#L66-L69.

Reply:
In my experience the use of global state is usually a very bad idea.

Reply:
yes, there is a trade off indeed, I'm ok keeping it configurable